lib/UI/list_item.dart

```diff
@@ -0,0 +1,165 @@
+import 'package:flutter/material.dart';
+
+
+class TaskItem extends StatefulWidget {
+  final Map<String, dynamic> task;
+  final DismissDirectionCallback onDismissed;
+  final VoidCallback onTap;
+  final ValueChanged<bool?> onCheckboxChanged;
+
+  const TaskItem({
+    required this.task,
+    required this.onDismissed,
+    required this.onTap,
+    required this.onCheckboxChanged,
+    super.key,
+  });
+
+  @override
+  _TaskItemState createState() => _TaskItemState();
+}
+
+class _TaskItemState extends State<TaskItem> {
+  @override
+  Widget build(BuildContext context) {
+    final task = widget.task;
+    return Container(
+      decoration: const BoxDecoration(
+        color: Colors.white,
+        borderRadius: BorderRadius.only(
+          topLeft: Radius.circular(10),
+          topRight: Radius.circular(10),
+        ),
+      ),
+      margin: const EdgeInsets.only(
+        left: 8,
+        right: 8,
+      ),
+      child: Dismissible(
+        key: Key(task['title']),
+        background: Container(
+          color: Colors.green,
+          padding: const EdgeInsets.only(left: 27),
+          child: const Row(
+            children: [
+              Icon(
+                Icons.check,
+                color: Colors.white,
+              ),
+            ],
+          ),
+        ),
+        secondaryBackground: Container(
+          color: Colors.red,
+          padding: const EdgeInsets.only(right: 27),
+          child: const Row(
+            mainAxisAlignment: MainAxisAlignment.end,
+            children: [
+              Icon(
+                Icons.delete,
+                color: Colors.white,
+              ),
+            ],
+          ),
+        ),
+        confirmDismiss: (direction) async {
+          widget.onDismissed(direction);
+          return direction == DismissDirection.endToStart;
+        },
+        child: ListTile(
+          leading: Checkbox(
+            value: task['completed'],
+            onChanged: widget.onCheckboxChanged,
+            side: BorderSide(
+              color:
+              (task["important"] == 1) ? Colors.red : Colors.grey.shade500,
+              width: 1.5,
+            ),
+            fillColor: MaterialStateProperty.resolveWith<Color>(
+                    (Set<MaterialState> states) {
+                  if (states.contains(MaterialState.selected)) {
```

Comment:
Само IDE (у меня Visual Studio Code) пишет, что использование MaterialStateProperty и MaterialState - уже "устарелые" методы. Лучше поменять на следующий код (функционал будет абсолютно таким же):
```suggestion
            fillColor: WidgetStateProperty.resolveWith<Color>(
                    (Set<WidgetState> states) {
                  if (states.contains(WidgetState.selected)) {
```